src/index.js

```diff
@@ -357,6 +374,20 @@ class Offline {
               firstCall = false;
             }
 
+            // Check for APIKey
+            if (_.includes(protectedRoutes, fullPath)) {
+              if ('x-api-key' in request.headers) {
+                const requestToken = request.headers['x-api-key'];
+                if (!_.includes(tokens, requestToken)) {
+                  debugLog(`method ${method} of function ${funName} token not found`);
+                  return reply({ errors: 'Wrong Token' }).code(403);
+                }
+              } else {
+                debugLog(`Missing x-api-key on private function ${funName}`);
+                return reply({ errors: 'Missing x-api-key on header'}).code(400);
+              }
```

Comment:
I've just tested the API keys using a real APIG deployement and it returns the same response with an incorrect x-api-key header or without it. The response is:

status code: 403
body: {"message":"Forbidden"}
headers we should include:
- Content-Type →application/json
- x-amzn-ErrorType: ForbiddenException